src/aap_eda/services/activation/engine/podman.py

```diff
@@ -143,9 +143,15 @@ def start(self, request: ContainerRequest, log_handler: LogHandler) -> str:
 
         try:
             LOGGER.info(f"Image URL is {request.image_url}")
-            if request.pull_policy == "Always" or not self._image_exists(
-                request.image_url,
+            image_existed = self._image_exists(request.image_url)
```

Comment:
it's still missing the logic for `never`, am I right?

Reply:
I disagree, that's incorrect. If we support `never` policy it must behave as expected. With never policy we should not try to pull the image never, it is supposed to be there. 
https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy

Reply:
@hsong-rh Podman actually accepts the pull policy [when running a container](https://docs.podman.io/en/v5.0.1/markdown/podman-run.1.html#pull-policy) in CLI or as [image operation](https://docs.podman.io/en/latest/_static/api.html#tag/images/operation/ImagePullLibpod) in the rest api (which is what the python client consumes)

In our current implementation we take care explicitly of the live cycle of the image so we need to implement the logic for all three options (only `never` would be missing)